airsonic-main/src/main/java/org/airsonic/player/service/search/IndexManager.java

```diff
@@ -110,9 +111,9 @@
     @Autowired
     private AlbumDao albumDao;
 
-    private EnumMap<IndexType, SearcherManager> searchers = new EnumMap<>(IndexType.class);
+    private Map<IndexType, SearcherManager> searchers = new ConcurrentHashMap<>();
```

Comment:
Two options here,
- either synchronize the EnumMap, 
  - `Collections.synchronizedMap(new EnumMap<>(...));`
- or switch to ConcurrentHashMap
  - I chose this, because I think it will be more performant. We'll look the efficiency of the EnumMap, but will allow concurrent access without needing for too much locking